zetaclient/chains/evm/signer/gas.go

```diff
@@ -0,0 +1,114 @@
+package signer
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+const (
+	minGasLimit = 100_000
+	maxGasLimit = 1_000_000
+)
+
+// Gas represents gas parameters for EVM transactions.
+//
+// This is pretty interesting because all EVM chains now support EIP-1559, but some chains do it in a specific way
+// https://eips.ethereum.org/EIPS/eip-1559
+// https://www.blocknative.com/blog/eip-1559-fees
+// https://github.com/bnb-chain/BEPs/blob/master/BEPs/BEP226.md (tl;dr: baseFee is always zero)
+//
+// However, this doesn't affect tx creation nor broadcasting
+type Gas struct {
+	Limit uint64
+
+	// This is a "total" gasPrice per 1 unit of gas.
+	Price *big.Int
+
+	// PriorityFee a fee paid directly to validators.
+	PriorityFee *big.Int
+}
+
+func (g Gas) validate() error {
+	switch {
+	case g.Limit == 0:
+		return errors.New("gas limit is zero")
+	case g.Price == nil:
+		return errors.New("max fee per unit is nil")
+	case g.PriorityFee == nil:
+		return errors.New("priority fee per unit is nil")
+	default:
+		return nil
+	}
+}
+
+// isLegacy determines whether the gas is meant for LegacyTx{} (pre EIP-1559)
+// or DynamicFeeTx{} (post EIP-1559).
+//
+// Returns true if priority fee is <= 0.
+func (g Gas) isLegacy() bool {
+	return g.PriorityFee.Sign() < 1
+}
+
+// makeGasFromCCTX creates Gas struct based from CCTX and priorityFee.
+func makeGasFromCCTX(cctx *types.CrossChainTx, logger zerolog.Logger) (Gas, error) {
```

Comment:
Shouldn't we rather use `newGasFromCCTX` terminology?

Reply:
https://github.com/zeta-chain/node/pull/2634/commits/4d6ed2cf32dcf27af7bc29271e7f67ed5bae5f5c